src/main/java/Duke.java

```diff
@@ -1,10 +1,163 @@
-public class Duke {
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+
+import java.io.IOException;
+
+/**
+ * A program that keeps track of a person's tasks.
+ */
+public class Duke extends Application {
+    private Storage storage;
+    private TaskList listOfTasks;
+    private Ui ui;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param filePath a String representing the file path
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        listOfTasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Method to run the program.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(listOfTasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException | IOException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+    /**
+     * Main method.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        new Duke("D:\\CS2103T\\duke\\docs\\duke.txt").run();
```

Comment:
You are using an absolute filepath (e.g. "D:/CS2103T/duke") but this will not apply for all users (e.g. if they have their CS2103T folder in "D:/Desktop/CS2103T") and will affect Duke's save function. Consider using a relative filepath (e.g. "./duke/doc/duke.txt") rather than an absolute one.